GUI.py

```diff
@@ -0,0 +1,581 @@
+        self.ShowMessage1(*self.data_prep())
+        while self.graph.toolbar.lx:
+            self.graph.toolbar.lx.pop(0).remove()
+        self.graph.canvas.draw()
+
+    def ShowMessage1(
+        self,
+        dat_X: np.ndarray = None,
+        dat_Y: np.ndarray = None,
+        wavelength: float = 9000.0,
+        n_wv_idx: float = None,
+        n_true: float = 3.54,
+    ) -> None:
+        """Draws the message box with info about sample thickness."""
+
+        self.psd, self.freq, true_freq, true_ind = fourier_analysis(dat_X, dat_Y)
+        msg = "Sample thickness is {h} um".format(
+            h=thickness(wavelength, 1 / true_freq, n_true)
+        )
+
+        dial = wx.MessageDialog(None, msg, "Info", wx.OK)
+        dial.ShowModal()
+        self.calc_btn.Enable(False)
+
+    def data_prep(self) -> Tuple[np.ndarray, np.ndarray, float, float, float]:
+        """Prepares data to be analysed."""
+        self.x = self.graph.toolbar.x[1:]
+        self.graph.toolbar.x[:] = []
+        self.x.sort()
+
+        try:
+            self.x != []
+        except:
+            print("Processing range not specified")
+
+        x = np.searchsorted(self.data[:, 0], [self.x[0]])[0]
+        y = np.searchsorted(self.data[:, 0], [self.x[1]])[0]
+        dat_X = self.data[x:y, 0]
+        dat_Y = self.data[x:y, 1]
+
+        wavelength = dat_X[int(dat_X.shape[-1] / 2)]
+        n_wv_idx = find_nearest(self.wvlngh, wavelength / 10000.0)
+        n_true = self.n_coef[n_wv_idx]
+
+        return dat_X, dat_Y, wavelength, n_wv_idx, n_true
+
+    def make_report(
+        self,
+        dat_X: np.ndarray = None,
+        dat_Y: np.ndarray = None,
+        wavelength: float = 9000.0,
+        n_wv_idx: float = None,
+        n_true=3.54,
+    ) -> None:
+        """Draws another panel covered with plots filled with additional info."""
+
+        self.calc_btn.Enable(False)
+
+        # dial = wx.ProgressDialog('Progress', 'Computation could take some time', maximum=100, parent=self)
+        # dial.Update(5)
+
+        # Moving average method
+        idx, peaks = find_peaks(dat_Y)
+        draw_data(
+            self.graphs.graphFour,
+            dat_X,
+            dat_Y,
+            scatter_x=dat_X[idx],
+            scatter_y=dat_Y[idx],
+            name="Peaks detection",
+        )
+        periodicity = pd.Series(dat_X[idx], index=dat_X[idx]).diff().dropna()
+        period = (
+            periodicity.rolling(window=10, center=True)
+            .mean()
+            .values[
+                int(
+                    periodicity.rolling(window=10, center=True).mean().values.shape[-1]
+                    / 2
+                )
+            ]
+        )
+        thickness(wavelength, period, n_true)
+
+        textstr = "\n".join(
+            (
+                r"$h=%.2f$ um" % (thickness(wavelength, period, n_true),),
+                r"$f =%.2f \frac{1}{\AA}$" % (1 / period,),
+                r"$\rho=%.2f\AA$" % (period,),
```

Comment:
I would recommend to use f-strings in python. By using f-strings you can get more elegant code.

Reply:
Why % is old school, you can find [here](https://realpython.com/python-f-strings/#old-school-string-formatting-in-python) =)